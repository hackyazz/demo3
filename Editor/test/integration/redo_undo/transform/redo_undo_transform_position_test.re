open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: transform position", () => {
    let sandbox = getSandboxDefaultVal();

    let _beforeEach = () => {
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode,
      );
    };

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "test simulate set currentSceneTreeNode",
      (
        RedoUndoTransformTool.simulateTwiceChangePosition(
          ~firstValue=11.25,
          ~secondValue=15.,
        ),
        _beforeEach,
        () => (),
      ),
      BuildComponentForCurryTool.buildMainEditorTransformComponent,
    );
  });