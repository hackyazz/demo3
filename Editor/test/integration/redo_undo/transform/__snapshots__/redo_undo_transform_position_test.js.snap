// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: transform position test simulate set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="11.25"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`redo_undo: transform position test simulate set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="11.25"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="15"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`redo_undo: transform position test simulate set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="11.25"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="15"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`redo_undo: transform position test simulate set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="11.25"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`redo_undo: transform position test simulate set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="position in local coordinate system"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="euler angle in local coordinate system"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
      title="scale in local coordinate system"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          title=""
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
